calico-cloud/image-assurance/scanners/registry-scanner.mdx

```diff
@@ -0,0 +1,150 @@
+- `AWS_ACCESS_KEY_ID`
+- `AWS_SECRET_ACCESS_KEY`
+- `AWS_REGION`
+
+The registry scanner currently scans one registry at a time. If you specify the Azure or AWS credentials, the Docker username and password will not be used.
+
+#### Required inputs
+
+The registry scanner requires:
+
+- `REGISTRY` - the registry you want to scan. (gcr.io)
+- `IMAGE_ASSURANCE_API_URL` - Get the URL in the Manager UI.
+- `IMAGE_ASSURANCE_API_TOKEN` - Get the token in the Manager UI.
+- Either the `DOCKER_USERNAME` and `DOCKER_PASSWORD`, Azure credentials, or AWS credentials.
+
+#### Optional inputs
+
+`REGISTRY_FILTER` - limits scanning time when you have thousands of repositories and images. Supports a comma-separated list.
+
+Example: for a gcr registry that contains:
+```
+gcr.io/prod-env/api
+gcr.io/staging-env/api
+gcr.io/dev/api
+```
+To filter out images in the dev "sub" registry, use:  `-e REGISTRY_FILTER=prod-env,staging-env`.
+
+### Scan images and send results to {{prodname}}
+
+Example run for a gcr registry with Docker Username and Password:
+
+```
+docker run -e REGISTRY=gcr.io -e IMAGE_ASSURANCE_API_URL=https://<clustId>-management.dev.calicocloud.io/bast
+ -e IMAGE_ASSURANCE_API_TOKEN=$TOKEN -e DOCKER_USERNAME=<username> -e DOCKER_PASSWORD=<password>
+ quay.io/tigera/image-assurance-registry-scanner:v1.10.4
+```
+
+Example run for an acr registry with Azure credentials:
+
+```
+docker run -e REGISTRY=your-org.azurecr.io -e IMAGE_ASSURANCE_API_URL=https://<clustId>-management.dev.calicocloud.io/bast
+ -e IMAGE_ASSURANCE_API_TOKEN=$TOKEN -e AZURE_CLIENT_ID=<azure-client-id> -e AZURE_CLIENT_SECRET=<azure-client-secret> -e AZURE_TENANT_ID=<azure-tenant-id>
+ quay.io/tigera/image-assurance-registry-scanner:v1.10.4
+```
+
+Example run for an ecr registry with AWS credentials:
+
+```
+docker run -e REGISTRY=gcr.io -e IMAGE_ASSURANCE_API_URL=https://<clustId>-management.dev.calicocloud.io/bast
+ -e IMAGE_ASSURANCE_API_TOKEN=$TOKEN -e AWS_ACCESS_KEY_ID=<aws-access-key> -e AWS_SECRET_ACCESS_KEY=<aws-secret-access-key> -e AWS_REGION=<aws-region>
+ quay.io/tigera/image-assurance-registry-scanner:v1.10.4
+```
+
+Example run when mounting the dockerfile:
+
+```
+docker run -e REGISTRY=gcr.io -e IMAGE_ASSURANCE_API_URL=https://<clustId>-management.dev.calicocloud.io/bast
+ -e IMAGE_ASSURANCE_API_TOKEN=$TOKEN -v ~/.docker/config.json/:/.docker/config.json quay.io/tigera/image-assurance-registry-scanner:v1.10.4
+```
+
+### Troubleshoot
+
+- For issues authenticating with the registry, verify that the registry credentials are correct.
+- For issues uploading the results to the {{prodname}}, check that the API token and URL are correct.
+- Because the scanner is not running continuously, it may take some time for results to display in the Image Assurance dashboard, especially if you have many images or large images.
+
+### Get registry scanner updates
+
+For the latest registry scanner versions, see [quay repository](https://quay.io/organization/tigera).
```

Comment:
I think we can keep this document up-to-date with the latest version of registry scanner using the {{imageassuranceversion}} variable from release to release.

So we can reword this part to be something like "To see previous versions of registry scanner ...".

Reply:
Fixed.